content-services/content-sources-backend
# Fixes 1579: update url on remote if url changed

## Summary

Supports updating url of a repository with snapshotting enabled.
Will update the remote to the new url as part of the snapshot
If the url is being updated, and a snapshot is in progress, we reject the request

## Testing steps

make this request:
```
POST http://localhost:8000/api/content-sources/v1.0/repositories/
Content-Type: application/json
x-rh-identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJqZG9lIn0sImludGVybmFsIjp7Im9yZ19pZCI6IjEyMyJ9fX0K

{"name":"sample zoo","url":"https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/","snapshot": true}
```

Then update the repo (change the uuid to match what is returned):

```
PATCH http://localhost:8000/api/content-sources/v1.0/repositories/1538f16f-324e-4d8e-95d4-483d73b3cebd
Content-Type: application/json
x-rh-identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJqZG9lIn0sImludGVybmFsIjp7Im9yZ19pZCI6IjEyMyJ9fX0K
```

Then check the snapshots table:

```
make db-cli-connect
select * from snapshots
```

you should see 2 created snapshots now.
